cli/cmd/main.go

```diff
@@ -712,6 +793,70 @@ func DataCommand(c *cli.Context) error {
 	}
 
 	filter := &datapb.Filter{}
+	if err := createDataFilter(c, filter); err != nil {
```

Comment:
I think in general it's better to return outputs* than modify inputs in place, so would suggest removing the filter parameter and just returning it instead:

```
filter, err := createDataFilter(c)
```

*A few reasons:

- If we modify an input, then we introduce a whole class of possible cases: what if the input is X/Y/Z/any variation here? This adds some complexity both because some of those inputs may be completely invalid (so require validation inside the function), and because it basically makes the "output" (in this case the implicit output being that modified input) a function of both the input state and the function implementation: instead of being able to treat the function like a black box that can be understood based solely on its name/signature/etc, a reader has to understand the initial input, the function, and how those two work together. Obviously this is super trivial in this case because the input is always the empty filter, but it leaves it open to these issues if that were to change in the future.
- The input is always the same (empty filter), so there's no need for the caller to need to provide it (and think about what they should provide/know it should be empty)

Reply:
done, lmk if this implementation makes sense to you